pkgs/desktops/lomiri/qml/lomiri-ui-toolkit/default.nix

```diff
@@ -0,0 +1,337 @@
+    done
+
+    # Install apicheck tool into bin
+    substituteInPlace apicheck/apicheck.pro \
+      --replace "\''$\''$[QT_INSTALL_LIBS]/lomiri-ui-toolkit" "$out/bin"
+
+    # Causes redefinition error with our own fortify hardening
+    sed -i '/DEFINES += _FORTIFY_SOURCE/d' features/lomiri_common.prf
+
+    # Reverse dependencies (and their reverse dependencies too) access the function patched here to register their gettext catalogues,
+    # so hardcoding any prefix here will make only catalogues in that prefix work. APP_DIR envvar will override this, but with domains from multiple derivations being
+    # used in a single application (lomiri-system-settings), that's of not much use either.
+    # https://gitlab.com/ubports/development/core/lomiri-ui-toolkit/-/blob/dcb3a523c56a400e5c3c163c2836cafca168767e/src/LomiriToolkit/i18n.cpp#L101-129
+    #
+    # This could be solved with a reference to the prefix of whoever requests the domain, but the call happens via some automatic Qt / QML callback magic,
+    # I'm not sure what the best way of injecting that there would be.
+    # https://gitlab.com/ubports/development/core/lomiri-ui-toolkit/-/blob/dcb3a523c56a400e5c3c163c2836cafca168767e/src/LomiriToolkit/i18n_p.h#L34
+    #
+    # Using /run/current-system/sw/share/locale instead of /usr/share/locale isn't a great
+    # solution, but at least it should get us working localisations
+    substituteInPlace src/LomiriToolkit/i18n.cpp \
+      --replace "/usr" "/run/current-system/sw"
+
+    # The code here overrides the regular QML import variables so the just-built modules are found & used in the tests
+    # But we need their QML dependencies too, so put them back in there
+    substituteInPlace export_qml_dir.sh \
+      --replace '_IMPORT_PATH=$BUILD_DIR/qml' '_IMPORT_PATH=$BUILD_DIR/qml:${qtQmlPaths}'
+
+    # statesaver test overrides XDG_RUNTIME_DIR with a custom path that it creates, in an attempt to ensure correct permissions
+    # Ironically this causes a warning by Qt about the permissions being wrong (755 instead of 700), which the test suite treats as an error
+    # Fix by setting the directory permissions
+    sed -i \
+      -e '/setenv("XDG_RUNTIME_DIR", testRuntimeDir.toUtf8()/a QFile::setPermissions(testRuntimeDir.toUtf8(), QFileDevice::ReadOwner | QFileDevice::WriteOwner | QFileDevice::ExeOwner);' \
+      tests/unit/statesaver/tst_statesaver.cpp
+
+    # These tests try to load Suru theme icons, but override XDG_DATA_DIRS / use full paths to load them
+    substituteInPlace \
+      tests/unit/visual/tst_visual.cpp \
+      tests/unit/visual/tst_icon.{11,13}.qml \
+      tests/unit/visual/tst_imageprovider.11.qml \
+      --replace '/usr/share' '${ubuntu-themes}/share'
+
+    # lomiri-mobile rename comes from where? maybe distro-specific change, maybe they use a different Suru?
+    substituteInPlace tests/unit/visual/tst_imageprovider.11.qml \
+      --replace 'lomiri-mobile' 'ubuntu-mobile'
```

Comment:
I don't know why they did that but it seems to originate from https://gitlab.com/ubports/development/core/lomiri-ui-toolkit/-/commit/976aeb5f40291044dcebb6d221821ef27bd44a78

Reply:
Seems like an unintentional change in a treewide commit hmm… I'll adjust the comment at least.

Reply:
Also submitted.